ktor-server/ktor-server-plugins/ktor-server-metrics/jvm/src/io/ktor/server/metrics/dropwizard/DropwizardMetrics.kt

```diff
@@ -105,3 +98,12 @@ public val DropwizardMetrics: ApplicationPlugin<DropwizardMetricsConfig> =
             }
         }
     }
+
+private fun PluginBuilder<DropwizardMetricsConfig>.callFinished(call: ApplicationCall) {
+    val routingMetrics = call.attributes.take(routingMetricsKey)
```

Comment:
It looks like 
```kotlin 
on(MonitoringEvent(Routing.RoutingCallStarted)) 
```
Isn't called for OPTIONS method http calls.

But
```kotlin
on(ResponseSent)
```
is called

Because of that there is always an IllegalStateException on every OPTIONS method call in this line.

Reply:
Hello, can you please provide a code as I tried to reproduce and there was no exception

Reply:
Looks like the issue happens when this plugin is used with CORS plugin:

Test
```kotlin
@Test
    fun `with Options call`() = testApplication {
        val testRegistry = MetricRegistry()
        install(DropwizardMetrics) {
            registry = testRegistry
            baseName = ""
        }
        install(CORS) {
            allowMethod(HttpMethod.Get)
            anyHost()
        }

        routing {
            get("/ok") {
                call.respond("OK")
            }
        }

        val response = client.options("/ok") {
            header("Access-Control-Request-Method", "GET")
            header("Origin", "https://ktor.io")
        }

        assertEquals(HttpStatusCode.OK, response.status)
        assertEquals(1, testRegistry.meter("/ok/(method:OPTIONS).200").count)
    }
```

DropwizardMetrics.kt
```kotlin
private fun PluginBuilder<DropwizardMetricsConfig>.callFinished(call: ApplicationCall) {
    try {
        val routingMetrics = call.attributes.take(routingMetricsKey)
        val status = call.response.status()?.value ?: 0
        val statusMeter =
            pluginConfig.registry.meter(name(pluginConfig.baseName, routingMetrics.name, status.toString()))
        statusMeter.mark()
        routingMetrics.context.stop()
    } catch (e: Exception) {
        println(e)
    }
}
```

Error:
```
java.lang.IllegalStateException: No instance for key AttributeKey: routing
```

It seems to me that CORS returns response here https://github.com/ktorio/ktor/blob/c1dbbcbed8ddf314aa33b4cb61457f3361ad6331/ktor-server/ktor-server-plugins/ktor-server-cors/jvmAndNix/src/io/ktor/server/plugins/cors/CORS.kt#L126
before Routing plugin had a chance to start.

Reply:
Added fallback in DropWizardMetric, if there was not called `on(MonitoringEvent(Routing.RoutingCallStarted))`